### DISCUSSION TITLE
Is it possible to flexibly update the header template, and still be able to auto generate?

### DISCUSSION THREAD

**Discussion by jhwan7** [davidquinn/psi-header#72] [Q&A] [answered]
Thank you for making this extension. I'm quiet enjoying using this extension, very customizable.

I recently came across a need to update my template, adding a new line to the existing template.

ex. (To give a simple example)
from
```
/*
 * Licensed Materials
 * Copyright (c) XXX 2024. All Rights Reserved.
 */
```
to
```
/*
 * Licensed Materials
 * Copyright (c) XXX 2024. All Rights Reserved.
 *  ---
 *  Additional Information
 */
```
So i added  the following 2 lines to the `"psi-header.templates"` in settings.json.

```
"---",
"Additional Information"
```

but subsequent saves does not apply this new template to the existing header that was on the file. Is there a way to make this happen?
Simply update the existing header with a new update template automatically?

**Comment by davidquinn** [comment, marked as answer]
I am sorry but it isn't. The problem is that it uses the initial characters in a line to identify the lines that have changed to map back to the template. There is no *__reliable__* way for it to map changes in the template to what is already in the file.

Whilst the current behaviour is not convenient for your use case, it does allow another significant use case - it allows the header to be manually edited and added to without significantly changing the behaviour of this extension.

I often delete the header and add a new one (via Ctrl-Alt-H, Ctrl-Alt-H) if I have made significant changes to the header as the end result should be what you want. However, if you have header comments enabled then these need to be saved and pasted back in after regenerating the header.

**Comment by jhwan7** [comment]
Thank you for the response.
I was doing it manually at the moment as well, but i guess that is the only way possible for now :)
